gateway/reverse_proxy_test.go

```diff
@@ -546,14 +546,20 @@ func TestSingleJoiningSlash(t *testing.T) {
 		a, b, want string
 	}{
 		{"foo", "", "foo"},
```

Comment:
Can we please add the following cases?
- `""`, `""`
- `"/"`, `""`
- `""`, `"/"`
- `"/"`, `"/"`